server.js

```diff
@@ -100,16 +167,19 @@ function runClient(client) {
   });
 
   client.on('upvote', function(data) {
-    for (var i in clients[client.id].upvoted) {
-      var hasAlreadyVoted = clients[client.id].upvoted[i].id == data.id && clients[client.id].upvoted[i].number == data.Number;
+    if (!clients[client.id]) {
+      return;
+    }
+    for (var x in clients[client.id].upvoted) {
+      var hasAlreadyVoted = clients[client.id].upvoted[x].id === data.id && clients[client.id].upvoted[x].number === data.number;
```

Comment:
I didn't know what this code was trying to do at first, so when I figured it out, I moved this conditional out into its own variable. This is a great way of making your code more readable; and when your code's more readable, it's more easily maintainable. By using a meaningful variable name, I don't even have to write a comment to explain myself.